pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -287,20 +289,21 @@ func (gb *GraphBuilder) IsSynced() bool {
 
 // Run sets the stop channel and starts monitor execution until stopCh is
 // closed. Any running monitors will be stopped before Run returns.
-func (gb *GraphBuilder) Run(stopCh <-chan struct{}) {
-	klog.Infof("GraphBuilder running")
-	defer klog.Infof("GraphBuilder stopping")
+func (gb *GraphBuilder) Run(ctx context.Context) {
+	logger := klog.FromContext(ctx)
+	logger.Info("Running", "component", "GraphBuilder")
+	defer logger.Info("Stopping", "component", "GraphBuilder")
 
 	// Set up the stop channel.
 	gb.monitorLock.Lock()
-	gb.stopCh = stopCh
+	gb.stopCh = ctx.Done()
```

Comment:
Nit: we could probably get rid of the `stopCh` entirely and replace it with context where needed, especially that I only found one place as such :wink: